lib/fast_cache/cache.rb

```diff
@@ -103,6 +105,7 @@ def empty?
     #
     # @return [self]
     def clear
+      @data.values.map(&:value).map(&@cleanup) if @cleanup
```

Comment:
This operation is unnecessarily inefficient:

1. `values` has to traverse the hash and create an array
1. `map` traverses the array and creates another array
1. `map` traverses that array and creates another array that is ignored

What you want here is a single `each_value` iteration.